nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -556,8 +580,19 @@ def load_checkpoint(self, checkpoint_path: Union[str, Path], load_optimizer_stat
             # Check whether to load optim states
             if load_optimizer_states:
                 checkpoint['optimizer_states'] = [self.optimizer_sharded_state_dict(is_loading=True)]
+
+            # Check if the model has a separate LM head stage
+            def is_lm_head_separate_stage(module=None, state_dict=None):
+                if module is None or state_dict is None:
+                    return False
+                if len(state_dict) > 1:
+                    return False
+                return isinstance(getattr(module.model.module, 'output_layer', None), ColumnParallelLinear)
+
+            patch_separate_lm_head_stage = is_lm_head_separate_stage(self.lightning_module, sharded_state_dict)
```

Comment:
L592 can be moved to the check of L594, which can avoid an unnecessary call for the case where param groups match.